### PR TITLE
Feature/ussd nurse rapidpro

### PR DESCRIPTION
**Pull request by OMosimege**

### CODE REVIEW

File: src/ussd_nurse_rapidpro.js [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    var GoNDOH = App.extend(function(self) {
+        App.call(self, 'state_start');
+        var utils = SeedJsboxUtils.utils;
+
+        //variables for services
+        var is;
+
+        self.init = function() {
+            // initialise services
+            is = new IdentityStore(
```
Comment (erikh360) [inline src/ussd_nurse_rapidpro.js:21]: I'm guessing we're probably not going to use this. Same for `IdentityStore` variable declared above.

File: src/ussd_nurse_rapidpro.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            return new ChoiceState(name, {
+                question: ("To register, your friend's info needs to be collected, stored and used. "+
+                            "They may receive messages on public holidays. Do they agree?"),
+                choices: [
+                    new Choice('yes', ('Yes')),
+                    new Choice('no', ('No')),
+                ],
+                next: function(choice) {
+                    if (choice.value === 'yes') {
+                        return 'state_agree_to_register';
```
Comment (erikh360) [inline src/ussd_nurse_rapidpro.js]: I see `state_weekly_messages` and `state_friend_register` both go to `state_agree_to_register` when the user agrees to share information, we will need to set a variable so we know who is registering when we get to `state_agree_to_register`.

File: test/ussd_nurse_rapidpro.test.js [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            tester = new AppTester(app);
+
+            tester
+                .setup.config.app({
+                    name: 'ussd_nurse_rapidpro',
+                    env: 'test',
+                    no_timeout_redirects: ['state_start'],
+                    channel: '*134*550*5#',
+                    services: {
+                        identity_store: {
```
Comment (erikh360) [inline test/ussd_nurse_rapidpro.test.js:24]: This is not necessary.

File: test/ussd_nurse_rapidpro.test.js [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+                            url: 'http://is/api/v1/',
+                            token: 'test IdentityStore'
+                        }
+                    },
+                })
+                .setup(function(api) {
+                    // add fixtures for services used
+                    fixtures_IdentityStore().forEach(api.http.fixtures.add); // 120 ->
```
Comment (erikh360) [inline test/ussd_nurse_rapidpro.test.js:32]: These fixtures mock the identity store so will probably be replaced with the companion app.

**Review by erikh360 — COMMENTED** [review]
Looks good so far! 🍰 

Few comments but mostly related to the identity store(seed) being replaced by the companion app. If you want to keep it there to replace later that is also fine, then you can ignore those comments.

Two things I would suggest for future PRs:
Add a description of what the PR contains.
When creating the branch include the Jira ticket number, for example `MOMZA-111-ussd-nurse-rapidpro-app`, it then shows all the branches and PRs on the ticket.

File: src/ussd_nurse_rapidpro.js (hunk 1 above)
Comment (OMosimege) [reply src/ussd_nurse_rapidpro.js:21]: yes, will remove that part after the companion app is complete, I just didn't know how to test without using the identity store. If you do have suggestions I'm open for help :)

File: src/ussd_nurse_rapidpro.js (hunk 2 above)
Comment (OMosimege) [reply src/ussd_nurse_rapidpro.js]: Ahh I see what you mean. But according to the flow diagram I was given, they both branch in and follow the same flow thereafter, which would make it hard if I differentiate them into 2 different variables. Or is it still wise to do so regardless?

File: test/ussd_nurse_rapidpro.test.js (hunk 3 above)
Comment (OMosimege) [reply test/ussd_nurse_rapidpro.test.js:24]: agreed, will be sure to note it out and have it replaced at a later stage thanks

File: test/ussd_nurse_rapidpro.test.js (hunk 4 above)
Comment (OMosimege) [reply test/ussd_nurse_rapidpro.test.js:32]: yes, they will be

File: src/ussd_nurse_rapidpro.js (hunk 2 above)
Comment (erikh360) [reply src/ussd_nurse_rapidpro.js]: It's fine that they go to the same state but when you get there you might want to know who is registering, the person dialling or a friend. So you could do something like:
`self.im.user.set_answer("friend_registration", true);`

**Review by erikh360 — APPROVED** [review]

**Pull request merged**